src/olympia/amo/cron.py

```diff
@@ -92,3 +94,31 @@ def write_sitemaps(section=None, app_name=None):
                 continue
             content = sitemap_object.render(app_name=_app_name, page=_page)
             sitemap_file.write(content)
+
+
+def record_metrics():
```

Comment:
I think we can do better than the comment to tie these two interfaces to each other. This is just an idea but lmk what you think:

1. define an ENUM representing the "queue" names. `ReviewQueueEnum`
2. use that enum in the [table dictionary](https://github.com/mozilla/addons-server/blob/c192a390b92b6a412ac81eba37e8c2ed309a10b2/src/olympia/reviewers/views.py#L418-L426) to specify relevant queue tables

3. use the same enum here to define the counts.

Currently it's not clear to me which key here maps to which table. I would need to read each table to find that. But if you have an enum connecting them, then I can trace the enum to the 2 interfaces. In a world where we have a rest interface the same can apply there as well.

Reply:
https://github.com/mozilla/addons-server/pull/22906/commits/c9076a48aaf1b77a1cf7de476e296a37c7b09288 goes in that direction; not completely finished yet, need to clean up a couple things.

I've switched back to the tables being the source of truth, even though the manager/queryset method is the actual source of truth, the way I'm handling this in each table should be more DRY.

Reply:
So, now:
- `reviewer_tables_registry` is where we specify queues
- Each queue table has a `name` and a `get_queryset()` method, with an optional `request` parameter to filter the queue according to the request/user.
- The `count()` optimization has been moved to its own queryset method on our base queryset class, in a dedicated method, `optimized_count()`.
- `record_metrics` simply loops over the registry, and for each queue in there, gets the queryset and calls `optimized_count()` on it. It re-uses the queue `name` as the key under which to record the value.

Reply:
This looks promising. I would still unify on the name property a bit more:
- move the formatted name passing the hard coded name string to the table class and just map the registry to produce the dict
- If we have the formatted name on the class, we could include this in the Metric instance and then dashboards can represent queue names exactly as they are in reviewer tools (ux benefit IMO)
- the cron should compile the querysets the same way the view does. Either use the tab, queue in registry.items() as in the view or ditch the tab entirely if we have the name. It's confusing to have both since they are not the same thing. e.g. is Metric.name referring to the tab key in the source of truth dict or the name of the table class?

Reply:
(I've made the querysets looping more consistent in https://github.com/mozilla/addons-server/pull/22906/commits/d7e318844145f8e8c7d7a9ea968ba6531ca1afce)

By formatted name, you mean the `title` ? We already have it on the classes, so that's easy to do. I see the benefit but I'm worried about it not being super stable, making data comparisons in the future harder if we decide to change the wording. The `name` is a completely internal value we are less likely to tweak.